app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class PostsController < ApplicationController
+  # before_action :signed_in?, only: [:new, :create]
```

Comment:
Why did you comment this line? that is a good way to do it. Just you can change the name to not confuse it with the `signed_in?` method I required to add.

Reply:
Makes sense, I uncommented it and refactored the code to use the #signed_in? private method, this uses the #logged_in? from the application helper included in the application controller, and redirect to the loggin page with flash message.